client/src/components/AdminSignup/index.js

```diff
@@ -2,12 +2,74 @@ import React, { Component } from 'react';
 import './AdminSignup.css';
 
 class AdminSignup extends Component {
+    constructor() {
+        super();
+        this.state = {
+            password: '',
+            confirmPassword: '',
+            hasStartedConfirmingPassword: false
+        }
+    }
+
     onSubmit = e => {
-        // TODO post the info to the server
+        const data = {
+            name: e.target.name.value,
+            email: e.target.email.value,
+            password: e.target.password.value
+        };
+        if (data.password !== e.target.confirmPassword.value) {
+            alert('Passwords must match before proceeding');
+            e.preventDefault();
+            return;
+        }
+        if (!data.password.length) {
+            alert('Password cannot be empty');
+            e.preventDefault();
+            return;
+        }
+        let headers = new Headers();
+        headers.append('Accept', 'application/json');
+        headers.append('Content-Type', 'application/json');
+        fetch('/admin', {
+            method: 'POST',
+            body: JSON.stringify(data),
+            headers
+        })
+        .then(resp => resp.json())
+        .then(json => {
+            // TODO handle parsed response (including errors)
+            debugger;
```

Comment:
Yeah. Dealing with users is something I haven't quite figured out and something that will take a lot to implement. At this point I'm thinking we add in redux and store the current user in the redux `store`. That'll be pretty involved and I want to keep commits small.